### DISCUSSION TITLE
tree-sitter playground tree-sitter-c version

### DISCUSSION THREAD

**Discussion by edogrigqv2** [tree-sitter/tree-sitter#2865] [Q&A]
I am working on a project where we are parsing C files. In tree-sitter-c there are lots of problems with preprocessing. I find tree-sitter online Playground to be the best version of tree-sitter-c. However I am having difficulty finding exact commit in the tree-sitter-c repo that is running on Playground. How can I find it?

**Comment by WillLillis** [comment]
[This discussion](https://github.com/tree-sitter/tree-sitter-bash/issues/148#issuecomment-1433918152) from tree-sitter-bash makes it seem like grammars are pulled from each repo's master branch, although it's not clear how often updates are made (maybe with each release?). There's also an [request](https://github.com/tree-sitter/tree-sitter/issues/2217) for commit hashes to be included in the playground which has been added to the [0.21.0 milestone](https://github.com/tree-sitter/tree-sitter/milestone/2). Hopefully the hashes get added soon :)
